tests/test_dimension.py

```diff
@@ -830,6 +831,148 @@ def test_implicit_dims(self):
 
         assert np.all(f.data == F)
 
+    def test_condition_lowering(self):
+        """
+        Test the lowering of a ConditionalDimension's condition.
+        This test makes an Operator that should indexify and lower the condition
+        passed in the Conditional Dimension
+        """
+
+        grid = Grid(shape=(3, 3))  # Define grid
+
+        # Define Function
+        g = Function(name='g', shape=grid.shape, dimensions=grid.dimensions)
+        g.data[:] = 1
+        x, y = grid.dimensions
+
+        ci = ConditionalDimension(name='ci', parent=y, condition=CondEq(g[x, y], 1))
+
+        f = Function(name='f', shape=grid.shape, dimensions=(x, ci))
+        Operator(Eq(f[x, ci], g[x, y])).apply()
+        assert np.all(f.data == 1)
+
+    @pytest.mark.parametrize('setup_rel, rhs, c1, c2, c3, c4', [
+        # Relation, RHS, c1 to c4 used as indexes in assert
+        (Lt, 3, 2, 4, 4, -1), (Le, 2, 2, 4, 4, -1), (Ge, 3, 4, 6, 1, 4),
+        (Gt, 2, 4, 6, 1, 4), (CondNe, 5, 2, 6, 1, 1)
```

Comment:
I don't think CondNe is necessary

Reply:
Removed. `Ne` will be added as soon as it is ready.